shared/middleware/helpers/null_pub_sub_api.rb

```diff
@@ -0,0 +1,11 @@
+# Interface to fake interactions with a PubSub system, like Pusher
+# Should respond to same messages as PusherApi
+
+class NullPubSubApi
```

Comment:
This comment is a little confusing; this is partly the fault of Ruby for not having a language notion of an interface.  Also, this is only the "publish" part of the pubsub API so your reader might be confused.

 How about the following:  
"Base class that defines an interface for publishing messages to a Pubsub system.  This class does nothing, derivedclasses should override publish with actual behavior"